src/main/java/com/google/sps/data/Game.java

```diff
@@ -15,6 +17,21 @@ public static Builder builder() {
   public abstract String creator();
   public abstract List<Question> questions();
 
+  public Map<String, Object> gameData() {
+
+    Map<String, Object> gameData = new HashMap<>();
+        gameData.put("creator", this.creator());
+        gameData.put("title", this.title());
+        gameData.put("numberOfQuestions", this.questions().size());
+    return gameData;
+  }
+
+  public Builder toBuilder() {
+    return new AutoValue_Game.Builder()
```

Comment:
I think AutoValue can generate this implementation for you. https://github.com/google/auto/blob/master/value/userguide/builders-howto.md#-initialize-a-builder-to-the-same-property-values-as-an-existing-value-instance

Reply:
Great! Thank you